mysite/ct/admin.py

```diff
@@ -1,19 +1,161 @@
 from django.contrib import admin
+from django.forms.models import ModelForm
+
 import ct.models
 
 
+class BaseForm(ModelForm):
+    """
+    Base class for admin forms.
+    """
+    def __init__(self, *args, **kwargs):
+        super(BaseForm, self).__init__(*args, **kwargs)
+
+        for field_name in ('addedBy', 'author'):
+            if field_name in self.fields and not self.initial.get(field_name):
+                self.initial[field_name] = self.current_user
+
+
+class ConceptForm(BaseForm):
+    """
+    Form for AdminConcept.
+    """
+    class Meta:
+        models = ct.models.Concept
+
+
+class ConceptGraphForm(BaseForm):
+    """
+    Form for AdminConceptGraph.
+    """
+    class Meta:
+        models = ct.models.ConceptGraph
+
+
+class LessonForm(BaseForm):
+    """
+    Form for AdminLesson.
+    """
+    class Meta:
+        models = ct.models.Lesson
+
+
+class ConceptLinkForm(BaseForm):
+    """
+    Form for AdminConceptLink.
+    """
+    class Meta:
+        models = ct.models.ConceptLink
+
+
+class UnitLessonForm(BaseForm):
+    """
+    Form for AdminUnitLesson.
+    """
+    class Meta:
+        models = ct.models.UnitLesson
+
+
+class UnitForm(BaseForm):
+    """
+    Form for AdminUnit.
+    """
+    class Meta:
+        models = ct.models.Unit
+
+
+class ResponseForm(BaseForm):
+    """
+    Form for AdminResponse.
+    """
+    class Meta:
+        models = ct.models.Response
+
+
+class StudentErrorForm(BaseForm):
+    """
+    Form for AdminStudentError.
+    """
+    class Meta:
+        models = ct.models.StudentError
+
+
+class CourseForm(BaseForm):
+    """
+    Form for AdminCourse.
+    """
+    class Meta:
+        models = ct.models.Course
+
+
+class CourseUnitForm(BaseForm):
```

Comment:
One note - we need to add fields attribute to modelform_factory because of RemovedInDjango18Warning: Creating a ModelForm without either the 'fields' attribute.